posts/2024-01-31-Performance-Doris_Amoakohene/index.qmd

```diff
@@ -0,0 +1,274 @@
+
+```{r,warning=FALSE,message=FALSE}
+
+library(atime)
+library(ggplot2)
+library(data.table)
+
+```
+
+```{r,warning=FALSE, message=FALSE}
+  tdir <- tempfile()
+  dir.create(tdir)
+  git2r::clone("https://github.com/Rdatatable/data.table", tdir )
+  
+```
+
+
+```{r, warning=FALSE, message=FALSE}
+
+atime.list.4200 <- atime::atime_versions(
+pkg.path=tdir,
+pkg.edit.fun=function(old.Package, new.Package, sha, new.pkg.path){
+      pkg_find_replace <- function(glob, FIND, REPLACE){
+        atime::glob_find_replace(file.path(new.pkg.path, glob), FIND, REPLACE)
+      }
+      Package_regex <- gsub(".", "_?", old.Package, fixed=TRUE)
+      Package_ <- gsub(".", "_", old.Package, fixed=TRUE)
+      new.Package_ <- paste0(Package_, "_", sha)
+      pkg_find_replace(
+        "DESCRIPTION", 
+        paste0("Package:\\s+", old.Package),
+        paste("Package:", new.Package))
+      pkg_find_replace(
+        file.path("src","Makevars.*in"),
+        Package_regex,
+        new.Package_)
+      pkg_find_replace(
+        file.path("R", "onLoad.R"),
+        Package_regex,
+        new.Package_)
+      pkg_find_replace(
+        file.path("R", "onLoad.R"),
+        sprintf('packageVersion\\("%s"\\)', old.Package),
+        sprintf('packageVersion\\("%s"\\)', new.Package))
+      pkg_find_replace(
+        file.path("src", "init.c"),
+        paste0("R_init_", Package_regex),
+        paste0("R_init_", gsub("[.]", "_", new.Package_)))
+      pkg_find_replace(
+        "NAMESPACE",
+        sprintf('useDynLib\\("?%s"?', Package_regex),
+        paste0('useDynLib(', new.Package_))
+    },
+
+  N=10^seq(1,20),
+  setup={ 
+    set.seed(108)
+    d <- data.table(
+      id3 = sample(c(seq.int(N*0.9), sample(N*0.9, N*0.1, TRUE))),
+      v1 = sample(5L, N, TRUE),
+      v2 = sample(5L, N, TRUE))
+  },
+  expr=data.table:::`[.data.table`(d, , (max(v1)-min(v2)), by = id3),
+"Before"="15f0598b9828d3af2eb8ddc9b38e0356f42afe4f",
+  "Regression"="6f360be0b2a6cf425f6df751ca9a99ec5d35ed93", 
+  "Fixed"="ba32f3cba38ec270587e395f6e6c26a80be36be6")
+```
+
+
+```{r, warning=FALSE,message=FALSE}
+plot(atime.list.4200)+
+  labs(title = "groupby with dogroups (R expression) performance regression")
+
+png("atime.list.4200.png")
+plot(atime.list.4200)+
+  labs(title = "groupby with dogroups (R expression) performance regression")
+dev.off()
+
+```
+The graph above  visualizes the  groupby operation with dogroups  issue. 
+
+Title: “Groupby with dogroups (R expression) performance regression.”
+
+Comparison: The graph compares the median time in milliseconds required to execute the operation before and after fixing a regression issue.
+
+1. X-Axis (N): Represents the size of the data (N) on a logarithmic scale.
+
+2.Y-Axis: Represents the median time in milliseconds (logarithmic scale).
+
+*Lines:*
+
+3. Red Line (“Before”): Indicates performance before fixing the regression.
+
+4. Green Line (“Fixed”): Shows improved performance after fixing.
```

Comment:
please change colors (red and green should not be used on same plot, for color blind people), try https://r-graph-gallery.com/38-rcolorbrewers-palettes.html or https://colorbrewer2.org/#type=qualitative&scheme=Accent&n=3

Reply:
okay